File: app/src/main/java/com/jordangellatly/coopervision/details/DetailPagerAdapter.java
package com.jordangellatly.coopervision.details;

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

public class DetailPagerAdapter extends FragmentPagerAdapter {
    private static int PAGE_COUNT = 2;
    private String tabTitles[] = new String[] {"Details", "Edit"};
    private Context context;

    public DetailPagerAdapter(FragmentManager fm, Context context) {
        super(fm);
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return DetailFragment.Companion.newInstance("Detail Fragment", 0);
            case 1:
                return EditFragment.newInstance("Edit Fragment", 1);
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        return tabTitles[position];
    }
}

File: app/src/main/java/com/jordangellatly/coopervision/details/DetailPagerAdapter.kt
package com.jordangellatly.coopervision.details

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class DetailPagerAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {

    override fun getItemCount() = PAGE_COUNT

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> DetailFragment.newInstance("Detail Fragment", 0)
            else -> EditFragment.newInstance("Edit Fragment", 1)
        }
    }

    companion object {
        private const val PAGE_COUNT = 2
    }
}
